### ISSUE TITLE
isEqualTo does not use withComperator if one of the arguments is null is not mentioned in the documentation

### ISSUE THREAD

**Issue by towi** [joel-costigliola/assertj-core#1120]
#### Summary
    
`assertThat(a).usingComparator(c).isEqualTo(b)` does not ask `c` if one of `a` or `b` is `null`. That itself is ok, but it must be mentioned in the documentation.

#### Example

```kotlin

    @Given("""^the data element is "([^"]*)"${'$'}""")
    fun `the data element is `(arg: String) {
        assertThat(rmessage!!.data?.elment)
                .usingComparator(blankOkComperator).isEqualTo(arg)
    }

    object blankOkComperator : Comparator<String> {
        override fun compare(o1: String?, o2: String?): Int {
            if(StringUtils.isBlank(o1) && StringUtils.isBlank(o2)) return 0
            if(StringUtils.isBlank(o1)) return -1
            if(StringUtils.isBlank(o2)) return 1
            return o1!!.compareTo(o2!!)
        }
    }

```

If `arg` is `""` -- which happens with Cucumber Examples -- then I would like that to be compared equal with `null`.

It should be mentionied in the documentation that `usingCompartor` is not suitable for that

It is mentioned in an internal function, but it should be visible at the interface level.

```java
  /**
   * Returns true if actual and other are equal according to {@link #comparator}, false otherwise.<br>
   * Handles the cases where one of the parameter is null so that internal {@link #comparator} does not have too.
   * 
   * @param actual the object to compare to other
   * @param other the object to compare to actual
   * @return true if actual and other are equal according to {@link #comparator}, false otherwise.
   */
  @Override
  @SuppressWarnings("unchecked")
  public boolean areEqual(Object actual, Object other) {
	if (actual == null) return other == null;
	// actual is not null
	if (other == null) return false;
	// neither actual nor other are null
	return comparator.compare(actual, other) == 0;
  }
```

**Comment by joel-costigliola** [comment]
I have created https://github.com/joel-costigliola/assertj-core/issues/1121 from your comment on SO - it is you right ? ;-)
TLDR, I think AssertJ should let the comparator handle null values.

**Comment by joel-costigliola** [comment]
closing this one since #1121 fixes it.
